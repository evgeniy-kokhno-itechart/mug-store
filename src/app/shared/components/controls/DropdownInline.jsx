import React from 'react';
import { PropTypes } from 'prop-types';
import Dropdown from './Dropdown';

const DropdownInline = ({
  name, label, error, value, options, defaultText, labelClasses, selectClasses, wrapperClasses, onChange,
}) => (
  <Dropdown
    name={name}
    label={label}
    error={error}
    value={value}
    options={options}
    defaultText={defaultText}
    labelClasses={labelClasses}
    selectClasses={selectClasses}
    wrapperClasses={`form-inline ${wrapperClasses}`}
    onChange={onChange}
  />
);

DropdownInline.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  error: PropTypes.string,
  value: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.shape({ id: PropTypes.string, name: PropTypes.string })).isRequired,
  defaultText: PropTypes.string,
  labelClasses: PropTypes.string,
  selectClasses: PropTypes.string,
  wrapperClasses: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

DropdownInline.defaultProps = {
  error: '',
  defaultText: '',
  labelClasses: '',
  selectClasses: '',
  wrapperClasses: '',
  value: '',
};

export default DropdownInline;
