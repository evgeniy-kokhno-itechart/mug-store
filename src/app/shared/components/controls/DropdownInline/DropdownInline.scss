.dropdown--inline {
    display: flex;
    align-items: center;
    flex-wrap: nowrap;
}