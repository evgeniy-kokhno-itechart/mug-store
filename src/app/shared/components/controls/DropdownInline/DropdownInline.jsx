import React from 'react';
import { PropTypes } from 'prop-types';
import Dropdown from '../Dropdown/Dropdown';
import './DropdownInline.scss';

const DropdownInline = ({
  name, label, error, value, options, defaultText, labelClasses, selectClasses, wrapperClasses, onChange,
}) => (
  <Dropdown
    name={name}
    label={label}
    error={error}
    value={value}
    options={options}
    defaultText={defaultText}
    labelClasses={labelClasses}
    selectClasses={selectClasses}
    wrapperClasses={`dropdown--inline ${wrapperClasses}`}
    onChange={onChange}
  />
);

DropdownInline.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  error: PropTypes.string,
  value: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.shape({ id: PropTypes.string, name: PropTypes.string })).isRequired,
  defaultText: PropTypes.string,
  labelClasses: PropTypes.string,
  selectClasses: PropTypes.string,
  wrapperClasses: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

DropdownInline.defaultProps = {
  error: '',
  defaultText: '',
  labelClasses: '',
  selectClasses: '',
  wrapperClasses: '',
  value: '',
};

export default DropdownInline;
