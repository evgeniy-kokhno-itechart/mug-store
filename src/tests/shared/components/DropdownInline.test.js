/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import DropdownInline from '../../../app/shared/components/controls/DropdownInline/DropdownInline';

configure({ adapter: new Adapter() });

describe('<DropdownInline />', () => {
  let DropdownInlineWrapper;

  beforeEach(() => {
    const onChange = jest.fn();
    DropdownInlineWrapper = shallow(
      <DropdownInline
        name="name"
        label="label"
        options={[{ id: '1', name: 'test name 1' }, { id: '2', name: 'test name 2' }]}
        onChange={onChange}
      />,
    );
  });

  it('renders properly with default props', () => {
    expect(DropdownInlineWrapper).toMatchSnapshot();
  });

  it('renders properly with error', () => {
    DropdownInlineWrapper.setProps({ error: 'test error' });
    expect(DropdownInlineWrapper).toMatchSnapshot();
  });

  it('renders properly with value provided', () => {
    DropdownInlineWrapper.setProps({ value: '2' });
    expect(DropdownInlineWrapper).toMatchSnapshot();
  });

  it('renders properly with defaultText', () => {
    DropdownInlineWrapper.setProps({ defaultText: 'test default text' });
    expect(DropdownInlineWrapper).toMatchSnapshot();
  });

  it('renders properly with labelClasses', () => {
    DropdownInlineWrapper.setProps({ labelClasses: 'testlabelclass' });
    expect(DropdownInlineWrapper).toMatchSnapshot();
  });

  it('renders properly with selectClasses', () => {
    DropdownInlineWrapper.setProps({ selectClasses: 'testselectclass' });
    expect(DropdownInlineWrapper).toMatchSnapshot();
  });

  it('renders properly with wrapperClasses', () => {
    DropdownInlineWrapper.setProps({ wrapperClasses: 'testwrapperclass' });
    expect(DropdownInlineWrapper).toMatchSnapshot();
  });
});
