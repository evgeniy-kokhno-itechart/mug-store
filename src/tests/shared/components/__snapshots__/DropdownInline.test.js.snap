// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownInline /> renders properly with default props 1`] = `
<Dropdown
  defaultText=""
  error=""
  label="label"
  labelClasses=""
  name="name"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "id": "1",
        "name": "test name 1",
      },
      Object {
        "id": "2",
        "name": "test name 2",
      },
    ]
  }
  selectClasses=""
  value=""
  wrapperClasses="dropdown--inline "
/>
`;

exports[`<DropdownInline /> renders properly with defaultText 1`] = `
<Dropdown
  defaultText="test default text"
  error=""
  label="label"
  labelClasses=""
  name="name"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "id": "1",
        "name": "test name 1",
      },
      Object {
        "id": "2",
        "name": "test name 2",
      },
    ]
  }
  selectClasses=""
  value=""
  wrapperClasses="dropdown--inline "
/>
`;

exports[`<DropdownInline /> renders properly with error 1`] = `
<Dropdown
  defaultText=""
  error="test error"
  label="label"
  labelClasses=""
  name="name"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "id": "1",
        "name": "test name 1",
      },
      Object {
        "id": "2",
        "name": "test name 2",
      },
    ]
  }
  selectClasses=""
  value=""
  wrapperClasses="dropdown--inline "
/>
`;

exports[`<DropdownInline /> renders properly with labelClasses 1`] = `
<Dropdown
  defaultText=""
  error=""
  label="label"
  labelClasses="testlabelclass"
  name="name"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "id": "1",
        "name": "test name 1",
      },
      Object {
        "id": "2",
        "name": "test name 2",
      },
    ]
  }
  selectClasses=""
  value=""
  wrapperClasses="dropdown--inline "
/>
`;

exports[`<DropdownInline /> renders properly with selectClasses 1`] = `
<Dropdown
  defaultText=""
  error=""
  label="label"
  labelClasses=""
  name="name"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "id": "1",
        "name": "test name 1",
      },
      Object {
        "id": "2",
        "name": "test name 2",
      },
    ]
  }
  selectClasses="testselectclass"
  value=""
  wrapperClasses="dropdown--inline "
/>
`;

exports[`<DropdownInline /> renders properly with value provided 1`] = `
<Dropdown
  defaultText=""
  error=""
  label="label"
  labelClasses=""
  name="name"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "id": "1",
        "name": "test name 1",
      },
      Object {
        "id": "2",
        "name": "test name 2",
      },
    ]
  }
  selectClasses=""
  value="2"
  wrapperClasses="dropdown--inline "
/>
`;

exports[`<DropdownInline /> renders properly with wrapperClasses 1`] = `
<Dropdown
  defaultText=""
  error=""
  label="label"
  labelClasses=""
  name="name"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "id": "1",
        "name": "test name 1",
      },
      Object {
        "id": "2",
        "name": "test name 2",
      },
    ]
  }
  selectClasses=""
  value=""
  wrapperClasses="dropdown--inline testwrapperclass"
/>
`;
